app/jobs/base.rb

```diff
@@ -102,30 +104,20 @@ def stop(exception:)
       end
 
       def self.raw_log(message)
+        begin
+          logger << message
+        rescue => e
+          Discourse.warn_exception(e, message: "Exception encountered while logging Sidekiq job")
+        end
+      end
+
+      def self.logger
         @@logger ||=
           begin
-            f = File.open "#{Rails.root}/log/sidekiq.log", "a"
-            f.sync = true
-            Logger.new f
+            log_path = "#{Rails.root}/log/sidekiq.log"
+            File.touch(log_path) if !File.exist?(log_path)
```

Comment:
The line is the only difference from 1670ffe82dbfa46f33df94312a8649ab8117d19b.

Reply:
Yup as long as they are created by the same process the permission will be the same. `Logger` doesn't do anything to alter the file's permission.